src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Persistence.cs

```diff
@@ -54,8 +54,12 @@ internal sealed partial class SyntaxTreeIndex : IObjectWritable
             //
             // We also want the checksum to change any time our serialization format changes.  If
             // the format has changed, all previous versions should be invalidated.
-            var projectChecksumState = await document.Project.State.GetStateChecksumsAsync(cancellationToken).ConfigureAwait(false);
-            var parseOptionsChecksum = projectChecksumState.ParseOptions;
+            var project = document.Project;
+
+            // This is the same logic the project itself uses here:
+            // https://github.com/dotnet/roslyn/blob/5e725e29186cc1b341b1a9b2baae73e34d245232/src/Workspaces/Core/Portable/Workspace/Solution/ProjectState_Checksum.cs#L48
+            var serializer = project.Solution.Workspace.Services.GetService<ISerializerService>();
+            var parseOptionsChecksum = ChecksumCache.GetOrCreate(project.ParseOptions, o => serializer.CreateChecksum(o, cancellationToken));
```

Comment:
the previous approach was simple and clearer.  however, it was about doubly expensive as it also required computed all the *other* checksums for the project, even though we only ever accessed the ParseOptions portion of it.